conda_build/variants.py

```diff
@@ -555,7 +555,10 @@ def find_used_variables_in_text(variant, recipe_text):
         conditional_regex = r"(?:^|[^\{])\{%\s*(?:el)?if\s*" + v_regex + r"\s*(?:[^%]*?)?%\}"
         # plain req name, no version spec.  Look for end of line after name, or comment or selector
         requirement_regex = r"^\s+\-\s+%s\s*(?:\s[\[#]|$)" % v_req_regex
-        all_res.extend([variant_regex, selector_regex, conditional_regex, requirement_regex])
+        if not selectors:
```

Comment:
This logic is counterintuitive to me.  I think it means:

Reply:
It should have been `if selectors`.

Reply:
Could you please document how you want this to behave?  I still don't trust myself to get this right, especially without some tests.